lib/adal.js

```diff
@@ -335,11 +335,17 @@ var AuthenticationContext = (function () {
         this._user = this._createUser(idtoken);
         return this._user;
     };
-
     /**
-     * If multiple requests for acquireToken are made for the same resource, we send only the first request and register the callbacks for all subsequent requests.
-     * When we process the response from AAD, this method calls all the callbacks with the error/token.
-     * @returns {User} user object
+     * This callback is displayed as part of the Requester class.
+     * @callback requestUserCallback
```

Comment:
You declared this class here but I don't see it being used here. Callbacks for both `acquiretoken` and `getUser` is same: requestCallback. I think you should define 2 classes: tokenCallback and userCallback. Set the callbacks for token related methods to be of type tokenCallback and for user related, set it to userCallback